### PR TITLE
Fixes #29514: Install puma-status with foreman-service package

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: packages/foreman/foreman/foreman.spec [hunk 1]
```diff
@@ -680,6 +680,7 @@ Meta Package to install dynflow sidekiq executor support
 Summary: Foreman systemd service support
 Group:  Applications/System
 Requires: %{?scl_prefix}rubygem(puma)
+Requires: %{?scl_prefix}rubygem(puma-status)
```
Comment (ekohl) [inline packages/foreman/foreman/foreman.spec:683]: We should make this auto-updateable:
```suggestion
# start specfile service Requires
Requires: %{?scl_prefix}rubygem(puma)
Requires: %{?scl_prefix}rubygem(puma-status)
# end specfile service Requires
```

**Comment by ehelms** [comment]
Deploys the script like the following on EL7:

```
#!/bin/bash
# Intended for a production environment only
/usr/bin/scl enable tfm "puma-status /run/foreman/puma.state"
```

File: packages/foreman/foreman/foreman.spec [hunk 2]
```diff
@@ -772,13 +776,17 @@ install -Dp -m0644 extras/systemd/%{dynflow_sidekiq_service_name}.service %{buil
 install -Dp -m0755 script/%{name}-debug %{buildroot}%{_sbindir}/%{name}-debug
 install -Dp -m0755 script/%{name}-rake %{buildroot}%{_sbindir}/%{name}-rake
 install -Dp -m0755 script/%{name}-tail %{buildroot}%{_sbindir}/%{name}-tail
+install -Dp -m0755 script/%{name}-puma-status %{buildroot}%{_sbindir}/%{name}-puma-status
 install -Dp -m0644 %{SOURCE3} %{buildroot}%{_sysconfdir}/logrotate.d/%{name}
 install -Dp -m0644 %{SOURCE4} %{buildroot}%{_sysconfdir}/cron.d/%{name}
 install -Dp -m0644 %{SOURCE5} %{buildroot}%{_tmpfilesdir}/%{name}.conf
 install -Dp -m0644 extras/systemd/%{name}.service %{buildroot}%{_unitdir}/%{name}.service
```
Comment (ekohl) [inline packages/foreman/foreman/foreman.spec:783]: On an unrelated note: we should `%exclude extras/systemd`

File: packages/foreman/foreman/foreman.spec [hunk 3]
```diff
@@ -772,13 +776,17 @@ install -Dp -m0644 extras/systemd/%{dynflow_sidekiq_service_name}.service %{buil
...
 install -Dp -m0644 extras/systemd/%{name}.socket %{buildroot}%{_unitdir}/%{name}.socket
 
+%if %{?scl:1}%{!?scl:0}
 sed -i '/^ExecStart/ s|/usr/bin/sidekiq \(.\+\)$|/usr/bin/scl enable tfm "sidekiq \1"|' %{buildroot}%{_unitdir}/%{dynflow_sidekiq_service_name}.service
+sed -i '/^puma-status/ s|puma-status \(.\+\)$|/usr/bin/scl enable tfm "puma-status \1"|' %{buildroot}%{_sbindir}/%{name}-puma-status
```
Comment (ekohl) [inline packages/foreman/foreman/foreman.spec:788]: Wouldn't this also work?

```suggestion
sed -i '/^puma-status/ s|\(puma-status .\+\)$|/usr/bin/scl enable tfm "\1"|' %{buildroot}%{_sbindir}/%{name}-puma-status
```

Comment (ehelms) [reply packages/foreman/foreman/foreman.spec:788]: Don't know, I can test it if this version prevents some bug or makes a big difference in the way it works?

**Pull request closed without merging**